src/GraphQL/Utilities/Visitors/Custom/PatternMatchingVisitor.cs

```diff
@@ -37,9 +37,16 @@ public override void VisitInterfaceFieldArgumentDefinition(QueryArgument argumen
     {
         // look for @pattern directive applied to the field argument or input field, and if found, use Validate to set the validation method
         var applied = fieldArgOrInputField.FindAppliedDirective("pattern");
-        if (applied?.FindArgument("regex")?.Value is not string regex)
+        var regexArgument = applied?.FindArgument("regex");
+        if (regexArgument == null)
             return null;
 
+        if (regexArgument.Value == null)
+            throw new ArgumentException("Pattern directive 'regex' argument must have non-null value");
+
+        if(regexArgument.Value is not string regex)
+            throw new ArgumentException("Pattern directive 'regex' argument must be of 'string' type");
```

Comment:
FYI, I think most of our XML comments and error messages end with a period.  (Not all yet though.)